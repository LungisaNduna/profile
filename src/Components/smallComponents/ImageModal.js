import React, { Component } from 'react'
import PropTypes from 'prop-types';
import GrowthPlan from '../../Resources/CareerPlan.png'

export class ImageModal extends Component {

  render() {
    // Render nothing if the "show" prop is false
    if(!this.props.show) {
      return null;
    }

    // The gray background
    const backdropStyle = {
      position: 'fixed',
      top: 0,
      bottom: 0,
      left: 0,
      right: 0,
      backgroundColor: 'rgba(0,0,0,0.3)',
      padding: 50
    };

    // The modal "window"
    const modalStyle = {
      backgroundColor: '#fff',
      borderRadius: 5,
      margin: '0 auto',
      padding: 20
    };

    return (
      <div className="backdrop" style={backdropStyle}>
        <div className="modal" style={modalStyle}>
          <h1 style={{marginTop: '0px'}}>Growth Plan</h1>
          <img id="growthPlan" src={GrowthPlan} alt="my personal growth plan"/>
          <span className="ImageSpaceError">The Picture resolution of the image will not work for this screen size</span>

          <div className="footer">
            <button onClick={this.props.onClose}>
              Close
            </button>
          </div>
        </div>
      </div>
    );
  }
}

ImageModal.propTypes = {
  onClose: PropTypes.func.isRequired,
  show: PropTypes.bool,
  children: PropTypes.node
};

    
    export default ImageModal;
