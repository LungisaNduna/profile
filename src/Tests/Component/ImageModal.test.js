import React from 'react';
import { shallow } from 'enzyme';
import ImageModal from '../../Components/smallComponents/ImageModal';

describe('Modal', () => {
  it('renders nothing when closed', () => {
    expect(shallow(
      <ImageModal onClose={jest.fn()}/>
    )).toMatchSnapshot();
  });

  it('renders when open', () => {
    expect(shallow(
      <ImageModal show={true} onClose={jest.fn()}/>
    )).toMatchSnapshot();
  });

  it('calls onClose when button is clicked', () => {
    const onClose = jest.fn();
    const wrapper = shallow(
      <ImageModal show={true} onClose={onClose}/>
    );

    wrapper.find('button').simulate('click');
    expect(onClose).toBeCalled();
  });
});