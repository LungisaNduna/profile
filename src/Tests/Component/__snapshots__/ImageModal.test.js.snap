// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders nothing when closed 1`] = `ShallowWrapper {}`;

exports[`Modal renders when open 1`] = `ShallowWrapper {}`;
